src/components/ClickableStyle/ClickableStyle.tsx

```diff
@@ -31,13 +34,45 @@ export type ClickableStyleProps<IComponent extends React.ElementType> = {
   /**
    * Available _color_ variations available for the ClickableStyle component
    */
-  status?: 'brand' | 'neutral' | 'success' | 'warning' | 'error';
+  status?: Status;
   /**
    * Available _shape_ variations available for the ClickableStyle component
    */
-  variant?: 'primary' | 'secondary' | 'icon' | 'link';
+  variant?: Variant;
 } & React.ComponentProps<IComponent>;
 
+const getPropCombinationIsValid = (variant: Variant, status: Status) => {
+  const nonPrimaryStatuses = ['neutral', 'success', 'warning'];
+  const nonLinkStatuses = ['success', 'warning', 'error'];
+
+  if (
+    (variant === 'primary' && nonPrimaryStatuses.includes(status)) ||
+    (variant === 'link' && nonLinkStatuses.includes(status))
+  ) {
+    return false;
+  }
+
+  return true;
+};
+
+const logInvalidPropComboWarning = (variant: Variant, status: Status) => {
+  const primaryStatuses = ['brand', 'error'];
+  const linkStatuses = ['brand', 'neutral'];
+
+  const getValidStatusesString = () => {
+    const validStatuses =
+      variant === 'primary' ? primaryStatuses : linkStatuses;
+
+    return `'${validStatuses[0]}' and '${validStatuses[1]}'`;
+  };
+
+  console.warn(
+    `*** Invalid prop combo warning ***:\n`,
+    `This component does not support using the '${status}' status with a '${variant}' variant.`,
+    `The '${variant}' variant can only be used with the ${getValidStatusesString} statuses.`,
+  );
```

Comment:
So I been throwing errors and not `console.warn`ing. Not sure which is the better approach

Reply:
I think it makes sense to throw an error if they're using something that is not supported/invalid and log a warning if it's something that's going to be deprecated but currently does work (because they're probably blocked by design on this one). I'll update this one to throw an error. Thanks!